src/frontend/src/optimizer/rule/index_delta_join.rs

```diff
@@ -90,9 +92,15 @@ impl Rule for IndexDeltaJoinRule {
 
         fn replace_across_exchange(plan: PlanRef, index_scan: PlanRef) -> PlanRef {
             if let Some(exchange) = plan.as_stream_exchange() {
-                exchange
-                    .clone_with_input(replace_across_exchange(exchange.input(), index_scan))
-                    .into()
+                if index_scan.distribution().satisfies(exchange.distribution()) {
```

Comment:
I think exchange's distribution will always be equivalent to index_scan. The current distribution derivation for indexes are wrong, so I think we still need to have the exchange to ensure correctness.

I'll fix this in later PRs.